sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -4476,6 +4476,18 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
       sql("select * from test_temp_view"),
       Row(1, 2, 3, 1, 2, 3, 1, 1))
   }
+
+  test("SPARK-40389: Don't eliminate a cast which can cause overflow") {
```

Comment:
Could you convert this into `sql-tests` under `sql/core/src/test/resources/sql-tests/inputs/ansi/`?

Reply:
The test case requires creating a real table. Otherwise, the rule "constant folding" will take effect...
In the `sql-tests`, creating a table and dropping it at the end in the golden file seems heavy.